tests/integration/pkg/cluster/clusters.go

```diff
@@ -165,6 +168,108 @@ func WaitFor(clients *clients.Clients, c *provisioningv1api.Cluster) (_ *provisi
 	return c, nil
 }
 
+func WaitForDelete(clients *clients.Clients, c *provisioningv1api.Cluster) (_ *provisioningv1api.Cluster, err error) {
+	defer func() {
+		if err != nil {
+			newC, newErr := clients.Provisioning.Cluster().Get(c.Namespace, c.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(newErr) {
+				newC = nil
+			} else if newErr != nil {
+				logrus.Errorf("failed to get cluster %s/%s to print error: %v", c.Namespace, c.Name, err)
+				return
+			}
+
+			machines, newErr := Machines(clients, c)
+			if newErr != nil {
+				logrus.Errorf("failed to get machines for %s/%s to print error: %v", c.Namespace, c.Name, err)
+			}
+
+			mgmtCluster, newErr := clients.Mgmt.Cluster().Get(c.Status.ClusterName, metav1.GetOptions{})
+			if apierrors.IsNotFound(newErr) {
+				mgmtCluster = nil
+			} else if newErr != nil {
+				logrus.Errorf("failed to get mgmt cluster %s/%s to print error: %v", c.Namespace, c.Name, err)
+			}
+
+			capiCluster, newErr := clients.CAPI.Cluster().Get(c.Namespace, c.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(newErr) {
+				capiCluster = nil
+			} else if newErr != nil {
+				logrus.Errorf("failed to get capi cluster %s/%s to print error: %v", c.Namespace, c.Name, err)
+			}
+
+			data, _ := json.Marshal(map[string]interface{}{
+				"cluster":     newC,
+				"mgmtCluster": mgmtCluster,
+				"capiCluster": capiCluster,
+				"machines":    machines,
+			})
+			err = fmt.Errorf("deletion wait failed on %s: %w", data, err)
+		}
+	}()
+
+	err = wait.Object(clients.Ctx, clients.Provisioning.Cluster().Watch, c, func(obj runtime.Object) (bool, error) {
+		c = obj.(*provisioningv1api.Cluster)
+		return !c.DeletionTimestamp.IsZero(), nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("cluster not deleted: %w", err)
+	}
+
+	machines, err := Machines(clients, c)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, machine := range machines.Items {
+		gvk := schema.FromAPIVersionAndKind(machine.Spec.InfrastructureRef.APIVersion, machine.Spec.InfrastructureRef.Kind)
+		gvr := schema.GroupVersionResource{
+			Group:    gvk.Group,
+			Version:  gvk.Version,
+			Resource: strings.ToLower(gvk.Kind) + "s",
```

Comment:
Is this acceptable to just append `s` here? I'm assuming we want to get plural name here.

Reply:
Since this is a test, this essentially will only be `podmachines`. I can change it to this explicitly if you think that is better.

Reply:
That's fine, thanks for the explanation.